### DISCUSSION TITLE
quick question about remote start and turn off time

### DISCUSSION THREAD

**Discussion by kurotwotails** [G-Two/homeassistant-subaru#110] [Q&A]
will this integration let my car remote start and run for more than 10 minutes or is the car going to shutoff after 10 minutes. does it allow me to turn my car on more then twice. thanks!

**Comment by G-Two** [comment]
The HA integration will use whatever settings that are specified in your remote start presets. The integration has no control or awareness of the parameters contained in the various presets (other than their names). 

The official mySubaru app only allows selection of 5 or 10 minute run times. The underlying [subarulink](https://github.com/G-Two/subarulink/) package (which is what the integration uses to interface with the Subaru API) also enforces 5 or 10 minute run times as the only valid options. You should be able to turn your car on again after it shuts off.

It is certainly possible to modify [subarulink constants](https://github.com/G-Two/subarulink/blob/fc1a6c2c151cf62be3b0be8a11a5d5429771b5ad/subarulink/const.py#L174) to remove the local bounds checking, but I don't know what would happen. The Subaru API backend or the vehicle might also do input validation. The typical response for invalid remote start parameters is that it gets fully processed by the Subaru API backend, is transmitted via telematics, and then the vehicle responds with a failure message. There is also a chance that there is no other input validation and that a value other than 5 or 10 will result in unexpected behavior. Or it might work.
